### ISSUE TITLE
代码能增加一点注释吗 看的头疼

### ISSUE THREAD

**Issue by yangohuang** [leventt/surat#4]

**Comment by leventt** [comment]
如果有你好奇的地方，我可以先在那里添加评论。 如果没有，我已经需要在有时间的时候更新它，但我不知道什么时候会更新。

**Comment by yangohuang** [comment]
hello, I have some questions with this part of the project:
`if mood is **not** None:
            out = torch.cat(
                (
                    out,
                    mood.view(
                        mood.view(-1, self.moodLen).size()[0], self.moodLen, 1, 1
                    ).expand(out.size()[0], self.moodLen, 64, 1)
                ),
                dim=1
            ).view(-1, 256 + self.moodLen, 64, 1)
        else:
            out = torch.cat(
                (
                    out,
                    torch.cat((
                        self.mood[moodIndex.chunk(chunks=1, dim=0)],
                        self.mood[(moodIndex + 1).chunk(chunks=1, dim=0)]
                    ), dim=0).view(
                        out.size()[0], self.moodLen, 1, 1
                    ).expand(out.size()[0], self.moodLen, 64, 1)
                ),
                dim=1
            ).view(-1, 256 + self.moodLen, 64, 1)` 

maybe the logic should be reversed ,which I think should be `if mood is  None:  XXX else:XXX`

thanks for your reply~

**Comment by leventt** [comment]
``` python
        if mood is not None:
            out = torch.cat(
                (
                    out,
                    mood.view(
                        mood.view(-1, self.moodLen).size()[0], self.moodLen, 1, 1
                    ).expand(out.size()[0], self.moodLen, 64, 1)
                ),
                dim=1
            ).view(-1, 256 + self.moodLen, 64, 1)
        else:
            out = torch.cat(
                (
                    out,
                    torch.cat((
                        self.mood[moodIndex.chunk(chunks=1, dim=0)],
                        self.mood[(moodIndex + 1).chunk(chunks=1, dim=0)]
                    ), dim=0).view(
                        out.size()[0], self.moodLen, 1, 1
                    ).expand(out.size()[0], self.moodLen, 64, 1)
                ),
                dim=1
            ).view(-1, 256 + self.moodLen, 64, 1)
```
no it shouldn't be reversed
it uses `self.mood` if `mood` is `None` and if it was passed with some value other than `None` then `mood` will be used for inference instead of the trainable parameter `self.mood`.
If  you were to reverse it, It would only use `mood` when it is passed as `None` and when it is passed with a meaningful value it wouldn't be used.
